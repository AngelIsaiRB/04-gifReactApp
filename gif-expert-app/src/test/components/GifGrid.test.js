const { GifGrid } = require("../../components/GifGrid");
import React from 'react'
import { shallow } from 'enzyme';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe("pruebas dentro de gifGrid",  ()=>{
    const category="hola world";

    test('should de hacr match con el snapshot', () => {

        useFetchGifs.mockReturnValue({
            data    : [],
            loading :true,
        });

        const wrapper = shallow(<GifGrid   category={category}/>);
        expect(wrapper).toMatchSnapshot();
    });

    test('should debe de mostrar items cuando carga imagenes useFetchsGisf', () => {
        const gifs =[{
            id:"ABC",
            url:"Https://localhostfnsofbsofudnf",
            title: "xx nada xx"
        }]
        useFetchGifs.mockReturnValue({
            data    : gifs,
            loading :false,
        });
        const wrapper = shallow(<GifGrid   category={category}/>);
        
        expect(wrapper).toMatchSnapshot();

    });
    
    

});

