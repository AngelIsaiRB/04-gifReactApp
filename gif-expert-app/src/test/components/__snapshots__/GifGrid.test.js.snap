// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas dentro de gifGrid should de hacr match con el snapshot 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    hola world
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    Loading . . .
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;

exports[`pruebas dentro de gifGrid should debe de mostrar items cuando carga imagenes useFetchsGisf 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__fadeIn"
  >
    hola world
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="xx nada xx"
      url="Https://localhostfnsofbsofudnf"
    />
  </div>
</Fragment>
`;
